cumulusci/tasks/apex/tests/test_apex_tasks.py

```diff
@@ -0,0 +1,348 @@
+                                "Cpu": 471,
+                                "Dml": 4,
+                                "Soql": 5,
+                            }
+                        ],
+                    },
+                }
+            ],
+        }
+
+        responses.add(
+            responses.GET, url, match_querystring=True, json=expected_response
+        )
+
+    def _mock_tests_complete(self):
+        url = (
+            self.base_tooling_url
+            + "query/?q=SELECT+Id%2C+Status%2C+"
+            + "ApexClassId+FROM+ApexTestQueueItem+WHERE+ParentJobId+%3D+%27"
+            + "JOB_ID1234567%27"
+        )
+        expected_response = {"done": True, "records": [{"Status": "Completed"}]}
+        responses.add(
+            responses.GET, url, match_querystring=True, json=expected_response
+        )
+
+    def _mock_run_tests(self):
+        url = self.base_tooling_url + "runTestsAsynchronous"
+        expected_response = "JOB_ID1234567"
+        responses.add(responses.POST, url, json=expected_response)
+
+    @responses.activate
+    def test_run_task(self):
+        self._mock_apex_class_query()
+        self._mock_run_tests()
+        self._mock_tests_complete()
+        self._mock_get_test_results()
+        task = RunApexTests(self.project_config, self.task_config, self.org_config)
+        task()
+        self.assertEqual(len(responses.calls), 4)
+
+
+@patch(
+    "cumulusci.tasks.salesforce.BaseSalesforceTask._update_credentials",
+    MagicMock(return_value=None),
+)
+class TestRunAnonApex(unittest.TestCase):
+    def setUp(self):
+        self.api_version = 42.0
+        self.global_config = BaseGlobalConfig(
+            {"project": {"api_version": self.api_version}}
+        )
+        self.task_config = TaskConfig()
+        self.task_config.config["options"] = {"apex": 'system.debug("Hello World!")'}
+        self.project_config = BaseProjectConfig(self.global_config)
+        self.project_config.config["project"] = {
+            "package": {"api_version": self.api_version}
+        }
+        keychain = BaseProjectKeychain(self.project_config, "")
+        self.project_config.set_keychain(keychain)
+        self.org_config = OrgConfig(
+            {"id": "foo/1", "instance_url": "example.com", "access_token": "abc123"},
+            "test",
+        )
+        self.base_tooling_url = "https://{}/services/data/v{}/tooling/".format(
+            self.org_config.instance_url, self.api_version
+        )
+
+    def _get_url_and_task(self):
+        task = AnonymousApexTask(self.project_config, self.task_config, self.org_config)
+        url = task.tooling.base_url + "executeAnonymous"
+        return task, url
+
+    @responses.activate
+    def test_run_anonymous_apex_status_fail(self):
+        task, url = self._get_url_and_task()
+        responses.add(responses.GET, url, status=418, body="I'm a teapot")
+        with assert_raises(SalesforceGeneralError) as cm:
```

Comment:
Is there a reason to use `nose.assert_raises` instead of `with self.assertRaises` from unittest? If no difference, I'd prefer the latter since it's in the stdlib.